include/circt/IR/CIRCTAttributes.td

```diff
@@ -0,0 +1,24 @@
+//===- CIRCTAttributes.td - Attributes for CIRCT -----------*- tablegen -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This file defines generic attributes for use by all CIRCT dialects.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef CIRCT_IR_CIRCTATTRIBUTES_TD
+#define CIRCT_IR_CIRCTATTRIBUTES_TD
+
+include "mlir/IR/OpBase.td"
+
+def LocationAttr : Attr<CPred<"$_self.isa<::mlir::LocationAttr>()">,
+  "location attribute">;
+
+def LocationArrayAttr : TypedArrayAttrBase<LocationAttr,
```

Comment:
This seems like it should be upstream.

Reply:
upstreamed here: https://reviews.llvm.org/D142148